spikewrap/data_classes/preprocessing.py

```diff
@@ -0,0 +1,58 @@
+import shutil
+from dataclasses import dataclass
+from typing import Dict
+
+import spikeinterface
+
+from ..utils import utils
+from .base import BaseUserDict
+
+
+@dataclass
+class PreprocessingData(BaseUserDict):
+    """
+    Dictionary to store SpikeInterface preprocessing recordings.
+
+    Details on the preprocessing steps are held in the dictionary keys e.g.
+    e.g. 0-raw, 1-raw-bandpass_filter, 2-raw_bandpass_filter-common_average
+    and recording objects are held in the value. These are generated
+    by the `pipeline.preprocess.run_preprocessing()` function.
+
+    The class manages paths to raw data and preprocessing output,
+    as defines methods to dump key information and the SpikeInterface
+    binary to disk. Note that SI preprocessing  is lazy and
+    preprocessing only run when the recording.get_traces()
+    is called, or the data is saved to binary.
+
+    Parameters
+    ----------
+    base_path : Union[Path, str]
+        Path where the rawdata folder containing subjects.
+
+    sub_name : str
+        'subject' to preprocess. The subject top level dir should
+        reside in base_path/rawdata/.
+
+    run_names : Union[List[str], str]
+        The SpikeGLX run name (i.e. not including the gate index)
+        or list of run names.
+    """
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        self._validate_rawdata_inputs()
+
+        self.sync: Dict = {}
+
+        for ses_name, run_name in self.preprocessing_sessions_and_runs():
+            utils.update(self.data, ses_name, run_name, {"0-raw": None})
+            utils.update(self.sync, ses_name, run_name, None)
```

Comment:
What is the use of `self.sync`?

Reply:
This is the synchonrisation channel that contains TTL pulse information. SI works in that the data used for sorting cannot contain the sync channel, however in `spikewrap` we want the sync channel as well. So we need to load it twice (once with once without sync). This could be made clearer though in the docs, maybe there is a cleaner way to write this.